applications/graph/MeshGraphNet/GNN.py

```diff
@@ -0,0 +1,114 @@
+import lbann
+from GNNComponents import MLP, GraphProcessor
+
+
+def input_data_splitter(input_layer,
+                        num_nodes,
+                        num_edges,
+                        in_dim_node,
+                        in_dim_edge,
+                        out_dim):
+  """ Takes a flattened sample from the Python DataReader and slices
+      them according to the graph attributes.
+  """
+
+  split_indices = []
+  start_index = 0
+  node_feature_size = num_nodes * in_dim_node
+  edge_feature_size = num_edges * in_dim_edge
+  out_feature_size =  num_nodes * out_dim
+
+  split_indices.append(start_index)
+  split_indices.append(split_indices[-1] + node_feature_size)
+  split_indices.append(split_indices[-1] + edge_feature_size)
+  split_indices.append(split_indices[-1] + num_edges)
+  split_indices.append(split_indices[-1] + num_edges)
+  split_indices.append(split_indices[-1] + out_feature_size)
+
+  sliced_input = lbann.Slice(input_layer, axis=0, slice_points=split_indices)
+
+  node_features = lbann.Reshape(lbann.Identity(sliced_input),
```

Comment:
Why are there `Identity` layers here? Wouldn't direct uses of `sliced_input` work?